web/ui/module/codemirror-promql/src/parser/parser.ts

```diff
@@ -300,6 +315,46 @@ export class Parser {
     }
   }
 
+  private checkSmoothedExpr(node: SyntaxNode): void {
+    // A smoothed expression is supposed to be used in the following functions only:
+    // rate, increase and delta.
+    // So we need to check that the parent is a FunctionCall and that the function name is one of the above.
+    const parent = node.parent?.parent;
+    if (!parent || parent.type.id !== FunctionCall) {
```

Comment:
Hmm what if there's no parent at all (should be ok), or if the parent is parentheses (doesn't make sense but should be ok), or if there's a future construct that allows modifying a range vector before feeding it into a function (like https://github.com/prometheus/prometheus/pull/14095)?

Maybe it is safer for this type of linting to only warn if the immediate parent (optionally after paren unwrapping) is a function, but one that isn't the right one?

Similar for the anchored checks.

Reply:
well from what I understood from https://github.com/prometheus/prometheus/pull/16457, this is not possible to use these keywords outside of specific function. So if there is no parent, then that's not correct.
And I think this is not possible to write a warning.